### PR TITLE
Merged changes from branch tymoteusz_gui.

### PR DESCRIPTION
**Pull request by Tymoteuszauke**
Merged changes from branch tymoteusz_gui. 
Added PositionUtils with methods that calculates position within array. Covered by tests.
BoardController contains temporary implementation of moveButton listener.

### CODE REVIEW

File: client/src/main/java/com/blackship/battlesheep/fx/controllers/BoardController.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import javafx.fxml.FXML;
+import javafx.geometry.HPos;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.TextArea;
+import javafx.scene.control.TextField;
+import javafx.scene.layout.GridPane;
+
+/**
+ * @author Mateusz Słaboński on 27.07.17
```
Comment (mhbratek) [inline client/src/main/java/com/blackship/battlesheep/fx/controllers/BoardController.java:14]: Check convention.

File: client/src/test/java/com/blackship/battlesheep/fx/controllers/utils/PositionUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        return new Object[][] {
+                {1, 0, 1},
+                {23, 2, 3},
+                {56, 5, 6},
+                {100, 10, 0}
+        };
+    }
+
+    @Test(dataProvider = "oneDimensionalPositionData")
+    public void testCalculateFromTwoDimensionalPosition(int expectedPosition, int givenX, int givenY) throws Exception {
```
Comment (mhbratek) [inline client/src/test/java/com/blackship/battlesheep/fx/controllers/utils/PositionUtilsTest.java:27]: In the future check naming convention.

File: client/src/main/java/com/blackship/battlesheep/fx/controllers/BoardController.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    public void handleMoveButtonAction(ActionEvent actionEvent) {
+
+        String moveValue = "";
+
+        moveValue = positionTextField.getText();
+        loggerInfoTextArea.appendText(moveValue + "\n");
+        positionTextField.clear();
+
+        int position = Integer.parseInt(moveValue);
+        int[] positionCoordinates = PositionUtils.calculateFromOneDimensionalPosition(position - 1, BOARD_SIZE);
```
Comment (mhbratek) [inline client/src/main/java/com/blackship/battlesheep/fx/controllers/BoardController.java:46]: Magic number 1

File: client/src/test/java/com/blackship/battlesheep/fx/controllers/utils/PositionUtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+import static org.testng.Assert.*;
+
+/**
+ * @author Mateusz Słaboński on 27.07.17
+ * @project
+ */
+public class PositionUtilsTest {
+
+    public static final int BOARD_SIZE = 10;
```
Comment (mhbratek) [inline client/src/test/java/com/blackship/battlesheep/fx/controllers/utils/PositionUtilsTest.java:14]: WHy public? :whale2:

**Review by mhbratek — APPROVED** [review]
Some minor changes.

File: client/src/test/java/com/blackship/battlesheep/fx/controllers/utils/PositionUtilsTest.java (hunk 2 above)
Comment (pawlusha) [reply client/src/test/java/com/blackship/battlesheep/fx/controllers/utils/PositionUtilsTest.java:27]: May be misleading to use oneDimensionalPositionData in test that calculatesFromTwoDimencionalPosition

**Review by pawlusha — COMMENTED** [review]
seems fine besides the naming convension in test. Make changes to that and you good to go

**Review by pawlusha — APPROVED** [review]

File: client/src/main/java/com/blackship/battlesheep/fx/controllers/BoardController.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+        Label label = new Label("X");
+        label.setStyle("-fx-font-size: 25px;");
```
Comment (piotr921) [inline client/src/main/java/com/blackship/battlesheep/fx/controllers/BoardController.java:49]: maybe some constant for style

File: client/src/main/resources/fx/BoardView.fxml [hunk 6]
```diff
@@ -8,9 +8,9 @@
 <?import javafx.scene.layout.Pane?>
 <?import javafx.scene.layout.RowConstraints?>
 
-<Pane maxHeight="600" maxWidth="1000" minHeight="-Infinity" minWidth="-Infinity" prefHeight="600.0" prefWidth="1000.0" xmlns="http://javafx.com/javafx/8.0.121" xmlns:fx="http://javafx.com/fxml/1" fx:controller="com.blackship.battlesheep.fx.BoardController">
+<Pane maxHeight="600" maxWidth="1000" minHeight="-Infinity" minWidth="-Infinity" prefHeight="600.0" prefWidth="1000.0" xmlns="http://javafx.com/javafx/8.0.121" xmlns:fx="http://javafx.com/fxml/1" fx:controller="com.blackship.battlesheep.fx.controllers.BoardController">
```
Comment (piotr921) [inline client/src/main/resources/fx/BoardView.fxml:11]: :+1:

File: client/src/test/java/com/blackship/battlesheep/fx/controllers/utils/PositionUtilsTest.java [hunk 7]
```diff
@@ -0,0 +1,46 @@
+package com.blackship.battlesheep.fx.controllers.utils;
+
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.*;
```
Comment (piotr921) [inline client/src/test/java/com/blackship/battlesheep/fx/controllers/utils/PositionUtilsTest.java:6]: import '*'

**Review by piotr921 — APPROVED** [review]

**Pull request closed without merging**